reactjs_yh/src/U2/hw1/todo.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from "react";
+var id_seed = 4;
+export default class todolist extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [
+        { title: "todo1", completed: false, id: 1 },
+        { title: "todo2", completed: false, id: 2 },
+        { title: "todo3", completed: false, id: 3 },
+      ],
+      todo: "new todo",
+    };
+  }
+  handleChange = (e) => {
+    const { value } = e.target;
+    this.setState((prevstate) => ({ ...prevstate, todo: value }));
+  };
+  handleSubmit = () => {
+    let newTodo = { title: this.state.todo, completed: false, id: id_seed++ };
+    let newTodos = this.state.todos.concat(newTodo);
+
+    this.setState((prevstate) => ({ ...prevstate, todos: newTodos }));
```

Comment:
here you can use spread operator to make it simpler, i.e., `{ todos: [...prevState.todos, newTodos] }`